tests/lib.rs

```diff
@@ -42,46 +50,87 @@ fn random_string(length: u64) -> String {
   (0..length).map(|_| (0x20u8 + (rand::random::<f32>() * 96.0) as u8) as char).collect()
 }
 
-/// Self Enryptor integration tests
-/*#[test]
-fn check_write() {
-  let mut se = SelfEncryptor::new(&mut my_storage as &mut Storage, datamap::DataMap::None);
-  se.write(&random_string(3), 5u64);
-  assert_eq!(se.len(), 8u64);
-}*/
-
 pub struct MyStorage {
-    name: Vec<u8>
+  temp_dir : TempDir
+}
+
+impl MyStorage {
+  pub fn new() -> MyStorage {
+    MyStorage { temp_dir: match TempDir::new("encrypt_storage") {
```

Comment:
This formatting is pretty confusing, `tempdir: TempDir::new(..).unwrap()` would suffice for test code. `unwrap` will print the error message on `Result`.